eagle-core/eagle-app/eagle-app-base/src/main/java/org/apache/eagle/app/spi/ApplicationProvider.java

```diff
@@ -72,7 +75,20 @@
      */
     void register(ModuleRegistry registry);
 
-    default Optional<HealthCheck> getAppHealthCheck(Config config) {
+    /**
+     * @param config application config.
+     * @return Application-specific managed health check.
+     */
+    default Optional<HealthCheck> getManagedHealthCheck(Config config) {
+        return Optional.empty();
+    }
+
+    /**
+     *
+     * @param envConfig server environment config.
+     * @return Server-level shared services.
+     */
+    default Optional<List<Service>> getSharedServices(Config envConfig) {
```

Comment:
Change#1: add `Optional<List<Service>> getSharedServices(Config envConfig) ` in `ApplicationProvider`

@qingwen220 @wujinhu